### ISSUE TITLE
setOnInsert() doesn't deduce @Property annotations

### ISSUE THREAD

**Issue by iAmcR** [MorphiaOrg/morphia#1932]
Given the class: 

```
public class SomeModel extends Model { 
 public String x;
 public String y;
 @Property("d")  
 public Date date;
}
```

doing an update() operation like: 
```
db.datastore.find(SomeModel.class)
.filter(eq("x", <x>), eq("y", <y>))
.update(
 setOnInsert(singletonMap("date", new Date()))
)
.execute(new UpdateOptions().upsert(true))
```

results in a document that sets the json property name to the literal 'date' instead of 'd' as defined by the `@Property` annotation.

Is this by design or is it something to be corrected or added to Morphia? 

** Please complete the following information: **
    Server Version: 5.0.6
    Driver Version: 4.2.3
    Morphia Version: 2.2.7

**Comment by evanchooly** [comment]
I'll call that a bug.  It's at least inconsistent naming elsewhere.  I'll look in to it.

**Comment by evanchooly** [comment]
I just pushed 2.2.8 which should fix your issue since 2.3.0 probably won't be out for a few more weeks.

**Comment by iAmcR** [comment]
@evanchooly Thank you. I'll check it out.
